routers/user/auth.go

```diff
@@ -885,14 +916,20 @@ func LinkAccountPostRegister(ctx *context.Context, cpt *captcha.Captcha, form au
 		ctx.ServerError("CreateUser", err)
 	}
 
+	// TODO LoginName should come from form.UserName... shouldn't it?
 	u := &models.User{
-		Name:        form.UserName,
-		Email:       form.Email,
-		Passwd:      form.Password,
-		IsActive:    !setting.Service.RegisterEmailConfirm,
-		LoginType:   models.LoginOAuth2,
-		LoginSource: loginSource.ID,
-		LoginName:   gothUser.(goth.User).UserID,
+		Name:     form.UserName,
+		Email:    form.Email,
+		Passwd:   form.Password,
+		IsActive: !setting.Service.RegisterEmailConfirm,
+	}
+
+	// This will link the account in such a way that it cannot be removed
+	// TODO why is this different from normal linking?
+	if setting.Service.AllowOnlyExternalRegistration {
```

Comment:
I don't think this is necessary. `setting.Service.AllowOnlyExternalRegistration` means user must register via external source. Since we have been in `LinkAccountPostRegister`. The check is unnecessary.